packages/node-integration-tests/utils/run-tests.ts

```diff
@@ -0,0 +1,60 @@
+/* eslint-disable no-console */
+import childProcess from 'child_process';
+import os from 'os';
+
+const testPaths = childProcess.execSync('jest --listTests', { encoding: 'utf8' }).trim().split('\n');
+
+let testsSucceeded = true;
+const numTests = testPaths.length;
+const fails: string[] = [];
+
+const threads = os.cpus().map(async (_, i) => {
```

Comment:
Nit: I think this makes it slightly clearer that two things are really happening, thread creation and thread use.

```suggestion
const threads = os.cpus()
threads.map(async (_, i) => {
```